editor/hooks/custom-class-name.native.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+import { assign } from 'lodash';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { hasBlockSupport } from '@wordpress/blocks';
+
+/**
+ * Filters registered block settings, extending attributes with anchor using ID
+ * of the first node.
+ *
+ * @param {Object} settings Original block settings.
+ *
+ * @return {Object} Filtered block settings.
+ */
+export function addAttribute( settings ) {
+	if ( hasBlockSupport( settings, 'customClassName', true ) ) {
+		// Use Lodash's assign to gracefully handle if attributes are undefined
+		settings.attributes = assign( settings.attributes, {
+			className: {
+				type: 'string',
+			},
+		} );
+	}
+
+	return settings;
+}
+
+/**
+ * Override props assigned to save component to inject anchor ID, if block
+ * supports anchor. This is only applied if the block's save result is an
+ * element and not a markup string.
+ *
+ * @param {Object} extraProps Additional props applied to save element.
+ * @param {Object} blockType  Block type.
+ * @param {Object} attributes Current block attributes.
+ *
+ * @return {Object} Filtered props applied to save element.
+ */
+export function addSaveProps( extraProps, blockType, attributes ) {
+	if ( hasBlockSupport( blockType, 'customClassName', true ) && attributes.className ) {
+		extraProps.className = classnames( extraProps.className, attributes.className );
+	}
+
+	return extraProps;
+}
+
+addFilter( 'blocks.registerBlockType', 'core/custom-class-name/attribute', addAttribute );
+addFilter( 'blocks.getSaveContent.extraProps', 'core/custom-class-name/save-props', addSaveProps );
```

Comment:
We should add here just another filter available on the web:
`addFilter( 'blocks.getBlockAttributes', 'core/custom-class-name/addParsedDifference', addParsedDifference );`

Also `cloneElement` and probably others need to be exposed.  Here is my working version https://github.com/WordPress/gutenberg/commit/3fdf7bdb06d1f15b452ef7c16884915a74d6fac0

Reply:
I exposed all `React` methods just in case we need them.

Yes, that one should help, too:
```
addFilter( 'blocks.getBlockAttributes', 'core/custom-class-name/addParsedDifference', addParsedDifference );
``